python/ray/tune/experiment.py

```diff
@@ -72,7 +73,7 @@ def __init__(self,
                  stop=None,
                  config=None,
                  trial_resources=None,
-                 repeat=1,
+                 num_samples=1,
```

Comment:
we should keep `repeat` in the arguments but in the arguments section of the docstring, note that it is deprecated.

Also, the `num_samples` docstring should be updated to make much more clearer. 

```
repeat (int): Deprecated and will be removed in future versions of Ray. Use `num_samples` instead.
num_samples (int): Number of times to sample from the hyperparameter space. Defaults to 1. If `grid_search` is provided as an argument, the grid will be repeated `num_samples` of times. 
```